apps/extension/server/src/server.ts

```diff
@@ -565,12 +559,39 @@ connection.onRequest('getMachineAtCursorPosition', ({ uri, position }) => {
   };
 });
 
+connection.onRequest('applyMachineEdits', ({ machineEdits }) => {
+  if (!displayedMachine) {
+    throw new Error(
+      '`applyMachineEdits` can only be requested when there is a displayed machine',
```

Comment:
Where does this error get displayed?

Reply:
Excellent question - it won't be displayed anywhere. This error has just been added to satisfy TS as it should be almost impossible to end up with it. I don't like "almost" in this sentence though and gonna try to look into providing stronger guarantees around this.